src/pages/billing/models/BillingProject.ts

```diff
@@ -15,18 +20,24 @@ export interface BillingProject {
   message?: string;
 }
 
-export interface AzureBillingProject extends BillingProject {
+export interface AzureBillingProject extends BaseBillingProject {
   cloudPlatform: 'AZURE';
   managedAppCoordinates: AzureManagedAppCoordinates;
   landingZoneId: string;
 }
 
-export interface GCPBillingProject extends BillingProject {
+export interface GCPBillingProject extends BaseBillingProject {
   cloudPlatform: 'GCP';
   billingAccount: string;
   servicePerimeter?: string;
 }
 
+export interface UnknownBillingProject extends BaseBillingProject {
+  cloudPlatform: 'UNKNOWN';
+}
+
+export type BillingProject = AzureBillingProject | GCPBillingProject | UnknownBillingProject;
```

Comment:
Using the union type instead of the base interface allows us to make use of type narrowing.

So we can do something like:
```ts
const billingProject: BillingProject = { ... }
if (billingProject.cloudPlatform === 'AZURE') {
  // TypeScript now narrows billingProject to AzureBillingProject
  billingProject.landingZoneId
}
```

As opposed to previously, this would require:
```ts
const billingProject: BaseBillingProject = { ... }
if (billingProject.cloudPlatform === 'AZURE') {
  // TypeScript has no additional information about billing project, so we have to cast to use fields defined on other types
  (billingProject as AzureBillingProject).landingZoneId
}
```

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions